pkg/controller/cluster/pools.go

```diff
@@ -0,0 +1,59 @@
+// This file is part of MinIO Operator
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	miniov1 "github.com/minio/operator/pkg/apis/minio.min.io/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/klog/v2"
+)
+
+// poolSSMatchesSpec checks if the statefulset for the pool matches what is expected and described from the Tenant
+func poolSSMatchesSpec(tenant *miniov1.Tenant, pool *miniov1.Pool, ss *appsv1.StatefulSet) (bool, error) {
+	// Verify Resources
+	updatePoolSS := false
+	if pool.Resources.String() != ss.Spec.Template.Spec.Containers[0].Resources.String() {
+		klog.V(4).Infof("resource requirements updates for pool %s", pool.Name)
+		updatePoolSS = true
+	}
```

Comment:
I don't know if this would be the case in Go, but in other languages what I've observed regarding object comparison using string serialization is that sometimes same object ends with attributes on different position and that will cause unnecessary updates

Reply:
this is the same comparison we had before, I just moved it to a new place. If we end up doing an update to the statefulset that has no changes, kubernetes will simply end up not doing anything, but so far in the tests this works every time